lib/hocon/config_error.rb

```diff
@@ -34,4 +34,7 @@ def initialize(message, cause)
 
   class ConfigNotResolvedError < Hocon::ConfigError::ConfigBugOrBrokenError
   end
+
+  class ConfigBadPathError < Hocon::ConfigError
```

Comment:
I think it might be worth porting over the error message logic for the `BadPath` class as well, since it's different than that for `ConfigException`.